xhyve/xhyve.go

```diff
@@ -603,22 +603,21 @@ func (d *Driver) extractKernelImages() error {
 	}
 
 	volumeRootDir := d.ResolveStorePath(isoMountPath)
-	vmlinuz := filepath.Join(volumeRootDir, "boot", d.Vmlinuz)
-	initrd := filepath.Join(volumeRootDir, "boot", d.Initrd)
 
-	log.Debugf("Extracting %s into %s", d.Vmlinuz, d.ResolveStorePath(d.Vmlinuz))
-	if err := mcnutils.CopyFile(vmlinuz, d.ResolveStorePath(d.Vmlinuz)); err != nil {
-		return err
-	}
-
-	log.Debugf("Extracting %s into %s", d.Initrd, d.ResolveStorePath(d.Initrd))
-	if err := mcnutils.CopyFile(initrd, d.ResolveStorePath(d.Initrd)); err != nil {
-		return err
-	}
+	defer func() {
+		log.Debugf("Unmounting %s", isoFilename)
+		err = hdiutil("detach", volumeRootDir)
+	}()
 
-	log.Debugf("Unmounting %s", isoFilename)
-	if err := hdiutil("detach", volumeRootDir); err != nil {
-		return err
+	for _, f := range []string{"vmlinuz64", "bzImage", "initrd", "initrd.img"} {
```

Comment:
Not sure I understand. Should I add vmlinux here as well? Or are you suggesting to use that instead of vmlinuz?